### PR TITLE
ARM express-route authorization implemented

### PR DESCRIPTION
**Pull request by MikhailTryakhov**
Unit tests implemented

### CODE REVIEW

**Comment by romangromov** [comment]
@MikhailTryakhov general comment: avoid unnecessary API calls (.showAuthorization() method call after .createOrUpdate). There is no need to fetch authorization object with GET after PUT, just use response body from PUT (.createOrUpdate) to output updated authorization object.

File: lib/commands/arm/network/expressRoute._js [hunk 1]
```diff
@@ -197,6 +197,151 @@ __.extend(ExpressRoute.prototype, {
...
+
+    self.interaction.formatOutput(circuitAuth, function (circuitAuth) {
+      if (circuitAuth === null) {
+        self.output.warn(util.format($('An express route circuit authorization with name "%s" not found in the circuit "%s" in resource group "%s"'), name, circuitName, resourceGroupName));
+      } else {
+        var resourceInfo = resourceUtils.getResourceInformation(circuitAuth.id);
+        self.output.nameValue($('Id'), circuitAuth.id);
+        self.output.nameValue($('Name'), circuitAuth.name);
+        self.output.nameValue($('Type'), resourceInfo.resourceType);
+        self.output.nameValue($('Use state'), circuitAuth.authorizationUseStatus);
```
Comment (romangromov) [inline lib/commands/arm/network/expressRoute._js]: This should be exactly `Use status`, not state

File: lib/commands/arm/network/expressRoute._js [hunk 2]
```diff
@@ -197,6 +197,151 @@ __.extend(ExpressRoute.prototype, {
...
+    } finally {
+      progress.end();
+    }
+    self.interaction.formatOutput(circuits, function (circuits) {
+      if (circuits.length === 0) {
+        self.output.warn($('No express route circuit authorizations found'));
+      } else {
+        self.output.table(circuits, function (row, circuitAuth) {
+          row.cell($('Name'), circuitAuth.name);
+          row.cell($('Use state'), circuitAuth.authorizationUseStatus);
```
Comment (romangromov) [inline lib/commands/arm/network/expressRoute._js]: This should be exactly `Use status`, not state

**Comment by MikhailTryakhov** [comment]
@romangromov comments about express route authorization from @nupurkot: "No, this value should be read only - only server sets it".

File: lib/commands/arm/network/expressRoute._js [hunk 3]
```diff
@@ -266,6 +399,24 @@ __.extend(ExpressRoute.prototype, {
...
+        self.output.warn(util.format($('An express route circuit authorization with name "%s" not found in the circuit "%s" in resource group "%s"'), name, circuitName, resourceGroupName));
+      } else {
+        var resourceInfo = resourceUtils.getResourceInformation(circuitAuth.id);
+        self.output.nameValue($('Id'), circuitAuth.id);
+        self.output.nameValue($('Name'), circuitAuth.name);
+        self.output.nameValue($('Type'), resourceInfo.resourceType);
+        self.output.nameValue($('Use status'), circuitAuth.authorizationUseStatus);
+        self.output.nameValue($('Authorization Key'), circuitAuth.authorizationKey);
+        self.output.nameValue($('Provisioning state'), circuitAuth.provisioningState);
+        self.output.nameValue($('Tags'), tagUtils.getTagsInfo(circuitAuth.tags));
```
Comment (romangromov) [inline lib/commands/arm/network/expressRoute._js]: tags for authorization not supported, please remove

File: lib/commands/arm/network/expressRoute._js [hunk 4]
```diff
@@ -197,6 +197,139 @@ __.extend(ExpressRoute.prototype, {
   },
 
   /**
+   * Circuit Authorization methods.
+   */
+  createAuthorization: function (resourceGroupName, circuitName, name, options, _) {
```
Comment (romangromov) [inline lib/commands/arm/network/expressRoute._js]: can we use `authName` instead of just `name`? we are following this convention everywhere

**Comment by romangromov** [comment]
@MikhailTryakhov overall - looks good, added two minor comments

**Comment by MikhailTryakhov** [comment]
Thank you @romangromov it's fixed
Hope it would be merged soon

**Comment by romangromov** [comment]
+@yugangw-msft @amarzavery, PR LGTM, please merge

**Pull request merged**